common/src/main/java/com/wynntils/services/itemfilter/ItemFilterService.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.services.itemfilter;
+
+import com.wynntils.core.components.Models;
+import com.wynntils.core.components.Service;
+import com.wynntils.core.components.Services;
+import com.wynntils.core.text.StyledText;
+import com.wynntils.models.items.WynnItem;
+import com.wynntils.services.itemfilter.filters.LevelItemFilter;
+import com.wynntils.services.itemfilter.filters.ProfessionItemFilter;
+import com.wynntils.services.itemfilter.type.ItemFilter;
+import com.wynntils.services.itemfilter.type.ItemFilterInstance;
+import com.wynntils.services.itemfilter.type.ItemSearchQuery;
+import com.wynntils.utils.type.ErrorOr;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+import java.util.Optional;
+import java.util.stream.IntStream;
+import net.minecraft.client.resources.language.I18n;
+import net.minecraft.world.item.ItemStack;
+
+public class ItemFilterService extends Service {
+    private final List<ItemFilter> filters = new ArrayList<>();
+
+    public ItemFilterService() {
+        super(List.of());
+        registerAllFilters();
+    }
+
+    /**
+     * Returns a filter for the given alias, or an error string if the alias does not match any filter.
+     * @param alias an alias of the filter
+     * @return the filter, or an error string if the alias does not match any filter
+     */
+    public ErrorOr<? extends ItemFilter> getFilter(String alias) {
+        Optional<ItemFilter> filterOpt = filters.stream()
+                .filter(filter ->
+                        filter.getAliases().contains(alias) || filter.getName().equals(alias))
```

Comment:
Flipping this or would feel a bit more natural to me, and assuming that the name is the most commonly used way to refer to the filter, it will also be slightly more efficient. (Not that it matters really)